test/mechanics/cell/Jacobian.cpp

```diff
@@ -85,6 +85,28 @@ BOOST_AUTO_TEST_CASE(Jacobian2Din3D)
     BOOST_CHECK_CLOSE(TriangleAreaViaJacobian(V(0, 4, 0), V(0, 0, 3), V(0, 4, 3)), 3. * 4 / 2, 1.e-10);
 }
 
+BOOST_AUTO_TEST_CASE(Jacobian2Din3DComputeNormal)
+{
+    Eigen::Vector3d p1(1., 0., 0.);
+    Eigen::Vector3d p2(0., 1., 0.);
+    Eigen::Vector3d p3(0., 0., 1.);
+
+    Eigen::MatrixXd B = NuTo::ShapeFunctions2D::DerivativeShapeFunctionsTriangleOrder1(Eigen::Vector2d::Zero());
+    Eigen::VectorXd coordinates = Eigen::VectorXd(9);
+    coordinates << p1, p2, p3;
+
+    NuTo::Jacobian jacobian(coordinates, B, 3);
+
+    Eigen::Vector3d basisVec1 = jacobian.Get().col(0);
+    Eigen::Vector3d basisVec2 = jacobian.Get().col(1);
+
+    Eigen::Vector3d normal = basisVec1.cross(basisVec2).normalized();
```

Comment:
Is seems to be equivalent to 
~~~cpp
Eigen::Vector3d normal = jacobian.Get().col(2);
~~~
right?

Reply:
That would be right if the returned jacobian had the magic 3rd column. Instead I chose to remove this feature because there were some reservations and now the jacobian is 2x3 and its inverse 3x2 and the (re)calculation of the normal has to be done elsewhere, e.g. in the boundaryloadfunction if needed.

Reply:
I am thinking about making this normal calculation more accessible.
- Maybe we could move the normal code to a free function - that only accepts _incomplete_ jacobians and throws otherwise
- Do you think the `jacobian.Get()` can be used for something else than calculating the normal? If that is unlikely, we could replace `.Get()` with `.Normal()`. Or simply add `.Normal()` to `CellIpData`?

Reply:
Usage of the Jacobian:
1) If you need a basis in tangential direction at some point you can use the two columns of the jacobian rather than computing one from the normal. In a conforming mesh a adjacent element will have the same basis vectors at a shared face.
2) If you want to get local coordinates from global ones in a higher order element you can use Newton iterations and the Jacobian will be the required derivative.